ibis/expr/types/arrays.py

```diff
@@ -426,23 +443,28 @@ def map(self, func: Callable[[ir.Value], ir.Value]) -> ir.ArrayValue:
         │ []                     │
         └────────────────────────┘
         """
-        name = next(iter(inspect.signature(func).parameters.keys()))
+        if isinstance(func, Deferred):
```

Comment:
these if/else blocks are shared between filter and map. Could extract them for reuse, but I thought that might just be more confusing.

Reply:
I think it's fine as you have it.